src/hocon_schema.erl

```diff
@@ -599,15 +599,29 @@ collect_envs() ->
 collect_envs(Ns) ->
     [begin
          [Name, Value] = string:split(KV, "="),
-         {Name, Value}
+         {Name, read_hocon_val(Value)}
      end || KV <- os:getenv(), string:prefix(KV, Ns) =/= nomatch].
 
+read_hocon_val("") -> "";
+read_hocon_val(Value) ->
+    try {ok, HoconVal} = hocon:binary(Value, #{}),
+        HoconVal
+    catch
+        _ : _ -> read_informal_hocon_val(Value)
+    end.
+
+read_informal_hocon_val(Value) ->
+    BoxedVal = "virtual_root=" ++ Value,
+    {ok, HoconVal} = hocon:binary(BoxedVal, #{}),
```

Comment:
raise exception when failed to parse this value

Reply:
fixed